zScanner/DomainModel/PageDomainModel.swift

```diff
@@ -40,8 +41,12 @@ struct PageDomainModel: Equatable {
 }
 
 extension PageDomainModel {
-    init(image: UIImage, index: Int, correlationId: String) {
-        self.init(id: UUID().uuidString, index: index, correlationId: correlationId, relativePath: "")
+    init(image: UIImage, index: Int, correlationId: String, description: String? = nil) {
```

Comment:
The thing with structs is that they create default initializer for oyu with all the variables you have specified in the struct itself. If you write your own initializer inside the struct, you will override the default one. 

if you will write initializer in the extension, you will keep the default init and new one to it. That's what I'm using here. I'm creating a new init which will provide unique ID. 

You just extended it by providing empty string `""` into the initializer and then setting up `self.description` later.  (I didn't notice this before)

Another thing you are doing is that you are storing `String` but your initializer accepts `String?` - optional. Don't do that. 
Either store optional or accept non-optional string on the init. Like this, you can store nil but when you read it, you receive empty string, which is something we don't want. Initialize and store the same value. and use directlu the `self.init()` on lie 45 instead of injecting it later on lines 49 and 50. 

The `image` here is used because it's a computed property with hidden extra logic, so it is not present in the initializer.